### PR TITLE
build: stress new tests for 5 minutes in CI

### PR DESCRIPTION
**Pull request by tamird**
Pretty crude for now, this runs new tests under stress for 5 minutes.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/10331)

### CODE REVIEW

File: pkg/cmd/github-pull-request-make/main.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+			pkgs[curPkg] = pkg
+		case newGoBenchmarkRE.Match(line):
+			pkg := pkgs[curPkg]
+			pkg.benchmarks = append(pkg.benchmarks, string(newGoBenchmarkRE.ReplaceAll(line, []byte(replacement))))
+			pkgs[curPkg] = pkg
+		}
+	}
+}
+
+func main() {
```
Comment (tschottdorf) [inline pkg/cmd/github-pull-request-make/main.go]: Comment on what happens in this file.

**Comment by jordanlewis** [comment]
Reviewed 5 of 8 files at r1, 8 of 8 files at r2.
Review status: 5 of 12 files reviewed at latest revision, 4 unresolved discussions.

---

*[pkg/cmd/github-pull-request-make/main.go, line 53 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQpzBrmv3DKaW_vuGF:-KVQpzBrmv3DKaW_vuGG:b-3sb5qt) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L53)):*
> ```Go
> }
> 
> func parseDiff(r io.Reader) (map[string]pkg, error) {
> ```

a quick doc on this function would be good as well.

---

*[pkg/cmd/github-pull-request-make/main.go, line 84 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQqJt0rcj5ZrbAELbC:-KVQqJt0rcj5ZrbAELbD:bmhn8vx) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L84)):*
> ```Go
> 		case newGoTestRE.Match(line):
> 			pkg := pkgs[curPkg]
> 			pkg.tests = append(pkg.tests, string(newGoTestRE.ReplaceAll(line, []byte(replacement))))
> ```

Does this properly work with subtests? Can you add a test for that if you haven't?

---

*[pkg/cmd/github-pull-request-make/main.go, line 159 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQq5sGD7mKrrd77YoK:-KVQq5sGD7mKrrd77YoL:b-49rcx2) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L159)):*
> ```Go
> 			fmt.Sprintf("PKG=./%s", name),
> 			fmt.Sprintf("TESTS=%s", tests),
> 			fmt.Sprintf("TESTFLAGS=-test.bench %s", benchmarks),
> ```

Seems potentially confusing to include this line if we don't have any benchmarks to test.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331)*

File: pkg/cmd/github-pull-request-make/main.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	var currentPull *github.PullRequest
+	for _, pull := range pulls {
+		if *pull.Head.SHA == sha {
+			currentPull = pull
+			break
+		}
+	}
+
+	resp, err := http.Get(*currentPull.DiffURL)
```
Comment (tschottdorf) [inline pkg/cmd/github-pull-request-make/main.go]: Interesting that you need to pull the diff from `GitHub`, I would've thought that you could get the base revision from the pull request's metadata and then diff against HEAD?

File: pkg/cmd/github-pull-request-make/main.go (hunk 1 above)
Comment (tamird) [inline pkg/cmd/github-pull-request-make/main.go]: Done.

**Comment by tamird** [comment]
Review status: 5 of 12 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[pkg/cmd/github-pull-request-make/main.go, line 53 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQpzBrmv3DKaW_vuGF:-KVQs7D1B0Y-U5olQlSm:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L53)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> a quick doc on this function would be good as well.
</details>
Done.

---

*[pkg/cmd/github-pull-request-make/main.go, line 84 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQqJt0rcj5ZrbAELbC:-KVQqy3YNOPAQSnDV_KN:bq8wws9) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L84)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> Does this properly work with subtests? Can you add a test for that if you haven't?
</details>
Depends on what you mean by "properly", but the answer is that this is completely unaware of subtests. If you add a new subtest, it will not be subject to stress testing.

---

*[pkg/cmd/github-pull-request-make/main.go, line 131 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQlwQAMgvMBD5_LB8r-r3-131:-KVQrAAF0p17V-yRWZ7G:b-gdsge1) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L131)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Interesting that you need to pull the diff from `GitHub`, I would've thought that you could get the base revision from the pull request's metadata and then diff against HEAD?
</details>
Yes, that would be nice, but TC doesn't know anything about pull requests. I've noted this in the top-level comment.

---

*[pkg/cmd/github-pull-request-make/main.go, line 159 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQq5sGD7mKrrd77YoK:-KVQrG3Bxw9ARI5_3OOl:bbnrpg) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L159)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> Seems potentially confusing to include this line if we don't have any benchmarks to test.
</details>
Meh, I think it's probably OK; `-test.bench -` doesn't seem _so_ bad.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331)*

**Comment by tschottdorf** [comment]
Crude or not, definitely good to have this. Only gave this a coarse review, but what I saw LGTM

---

Reviewed 1 of 8 files at r1, 7 of 8 files at r2, 5 of 7 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[pkg/cmd/github-pull-request-make/main.go, line 131 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQlwQAMgvMBD5_LB8r-r3-131:-KVQsesjmGCmuML8cxR3:bnvwnca) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L131)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Yes, that would be nice, but TC doesn't know anything about pull requests. I've noted this in the top-level comment.
</details>
Sorry, I mean that it seems that querying the pull request API should give you the base sha and that should be enough to get you the diff locally? It looks like you're pulling the diff from Github. Don't feel strongly (since we need to query the API anyway), just curious.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-:-KVQt6LDwCFUo-bCvlD1:b2n38j9)*

**Comment by jordanlewis** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[pkg/cmd/github-pull-request-make/main.go, line 131 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQlwQAMgvMBD5_LB8r-r3-131:-KVQtAejyOmzUcvIA-2R:b1uy365) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L131)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Sorry, I mean that it seems that querying the pull request API should give you the base sha and that should be enough to get you the diff locally? It looks like you're pulling the diff from Github. Don't feel strongly (since we need to query the API anyway), just curious.
</details>
TC doesn't know anything more or less about pull requests than any other build system before they added API integration with GitHub. I agree with Tobi - you could get the base revision from the pull request's metadata using an API call from the build script, diff against head and send the results to this program.

On the other hand I don't see a particularly compelling reason to favor that approach over this one - they seem more or less equivalent.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[pkg/cmd/github-pull-request-make/main.go, line 131 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331#-KVQlwQAMgvMBD5_LB8r-r3-131:-KVQuJPb3_t0mc3MwK6j:bqyhujs) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f84ad457ec8ad7ea9b66764938c4dbaf5f069dd/pkg/cmd/github-pull-request-make/main.go#L131)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> TC doesn't know anything more or less about pull requests than any other build system before they added API integration with GitHub. I agree with Tobi - you could get the base revision from the pull request's metadata using an API call from the build script, diff against head and send the results to this program.
> 
> On the other hand I don't see a particularly compelling reason to favor that approach over this one - they seem more or less equivalent.
</details>
Right, that's true; I could get the diff locally but there's no obvious to prefer that approach (and it's perhaps susceptible to missing refs? idk, decided not to mess with that just now).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10331)*

**Pull request merged**